### PR TITLE
Convert most OSS plugins from integTest to [yaml | java]RestTest or internalClusterTest

### PR DESCRIPTION
**Pull request by jakelandis**
For all OSS plugins (except repository-* and discovery-*) `integTest` 
task is now a no-op and all of the tests are now executed via a `test`,
`yamlRestTest`, `javaRestTest`, or `internalClusterTest`.  

This commit does NOT convert the discovery-* and repository-* since they 
are bit more complex then the rest of tests and this PR is large enough.
Those plugins will be addressed in a future PR(s). 

This commit also fixes a minor issue that did not copy the rest api
for projects that only had YAML TEST tests.  

related: #56841

### CODE REVIEW

File: plugins/analysis-icu/build.gradle
Comment (jakelandis) [inline plugins/analysis-icu/build.gradle:29]: this is needed since the yamlRestRest does not have access to that class

Comment (rjernst) [reply plugins/analysis-icu/build.gradle:29]: Maybe this should be explicit, applying the signature to `forbiddanApisMain` and `forbiddenApisTest`, rather than all CheckForbiddenApis tasks? We are likely to continue adding more source sets in the future, so having configuration apply to all source sets would be problematic.

File: plugins/analysis-kuromoji/build.gradle
Comment (rjernst) [inline plugins/analysis-kuromoji/build.gradle:36]: Why is this being disabled?

File: plugins/analysis-nori/build.gradle
Comment (rjernst) [inline plugins/analysis-nori/build.gradle:36]: Ditto. If we need to disable a test, can we have a comment explaining why it is disabled? Someone coming along in the future will have no reason whether this can/should be re-enabled.

**Review by rjernst — COMMENTED** [review]
I just have a few comments

File: plugins/analysis-kuromoji/build.gradle
Comment (mark-vieira) [reply plugins/analysis-kuromoji/build.gradle:36]: We need a plan for ripping the creation of the `integTest` task out from the `PluginBuildPlugin`. Otherwise this is going to be necessary for any plugin project that uses only yaml tests. Or tweak our test conventions logic so it doesn't complain about this scenario.

Comment (jakelandis) [reply plugins/analysis-kuromoji/build.gradle:36]: I found a need to implement a `javaRestTest` as part of converting these plugins, so (without hacks) can't finish my conversion to `yamlRestTest` .. Between `javaRestTest`, `yamlRestTest`, and `internalClusterTest` there is no need for `integTest` to exist anymore.  I plan to rip out `integTest` once all of these are implemented.  It won't take long (but my attention is split among a few things at the moment).

File: plugins/analysis-nori/build.gradle
Comment (jakelandis) [reply plugins/analysis-nori/build.gradle:36]: This isn't really disabling a test ... it is to avoid the testing conventions from complaining.  I would prefer to tweak the testing conventions instead of a lot of redunant comments. I can do that in a separate PR (and remove these in this PR)

**Comment by jakelandis** [comment]
@rjernst @mark-vieira - https://github.com/elastic/elasticsearch/pull/59939 has been submitted that introduces the javaRestTest plugin and converts the modules over. That PR also conditionally disables the `integTest` task so once that gets merged, and I bring those changes to this PR, i will update this PR to remove usages of `integTest` too.

File: plugins/analysis-icu/build.gradle
Comment (jakelandis) [reply plugins/analysis-icu/build.gradle]: :+1: done [`5c75412` (#59444)](https://github.com/elastic/elasticsearch/pull/59444/commits/5c75412f830b1ea938bbfee39b2a20c4bc6f90d8) for the main source set (probably not the extra config noise for the test and internalTestCluster source sets)

**Comment by jakelandis** [comment]
hold off on reviewing... now that #59939 has been merged and integrated here, there is still some more work on this PR.

**Comment by jakelandis** [comment]
@rjernst @mark-vieira - this is ready for review. Despite the # of files changes, this should be pretty straight forward review. Most of the changes are just moving files around and implementing the new plugin(s). These are the "easy" conversions and have will have additional PRs for the more involved conversion.

File: plugins/analysis-icu/build.gradle
Comment (mark-vieira) [inline plugins/analysis-icu/build.gradle:29]: What's the purpose of this of this intermediary variable?

File: plugins/examples/custom-settings/build.gradle
Comment (mark-vieira) [inline plugins/examples/custom-settings/build.gradle:30]: Is there a reason this needs to be `all` and not just `yamlRestTest`? Isn't there just one test cluster in play here?

File: plugins/examples/rest-handler/build.gradle
Comment (mark-vieira) [inline plugins/examples/rest-handler/build.gradle:37]: Should be `dependsOn sourceSets.javaRestTest.runtimeClasspath`.

File: plugins/examples/security-authorization-engine/build.gradle
Comment (mark-vieira) [inline plugins/examples/security-authorization-engine/build.gradle:16]: Why is this necessary here but not done by default in the java rest test plugin?

Since this is an example project we should use idiomatic patterns as much as possible. If this is required because of some kind of nasty test implementation we should refactor this.

@rjernst thoughts? As I understand it a Java REST test should be no different than a YAML one in that the only interactions should be via the REST API, therefore there should be no need for plugin code itself to be on the test classpath.

File: plugins/transport-nio/build.gradle
Comment (mark-vieira) [inline plugins/transport-nio/build.gradle:30]: I thought we ditched the need for this?

**Review by mark-vieira — APPROVED** [review]
Few minor comments otherwise 👍

File: plugins/examples/custom-settings/build.gradle
Comment (jakelandis) [reply plugins/examples/custom-settings/build.gradle:30]: I figured the best example would be to use `all` , I think that should be the go-to configuration unless you need task specific config. (given the ease and cache behavior). Not a strong opinion however.

File: plugins/analysis-icu/build.gradle
Comment (jakelandis) [reply plugins/analysis-icu/build.gradle:29]: it made sense when i was configuring multiple forbidden[name] ... but now not so much...will fix.

File: plugins/transport-nio/build.gradle
Comment (jakelandis) [reply plugins/transport-nio/build.gradle:30]: This one is special such neither javaRestTest or yamlRestTest is applied here (only internalClusterTest). The auto-disable only works from the java/yamlRestTest application since there are some other projects with existing `internalClusterTest` and don't want to auto disable integTest for those. These should be pretty rare, and can be deleted once integTest is gone.

File: plugins/examples/security-authorization-engine/build.gradle
Comment (jakelandis) [reply plugins/examples/security-authorization-engine/build.gradle:16]: The examples are not exactly idomatic to start with, they could use some updating. In this case the test really does need to see the main classes, which should probably be converted to a `ESSingleNodeTestCase` or the tests written as pure black box ?  I can log an issue, but probably outside the scope of this change.

Comment (mark-vieira) [reply plugins/examples/security-authorization-engine/build.gradle:16]: Yeah, I don't like the idea of `GradleUtils` being used in a plugin example. That's not intended to be "public". Worst case if the tests are structure such that they strictly need those classes, we should just extend the source set in the build script explicilty rather than leverage this utility method.

Comment (jakelandis) [reply plugins/examples/security-authorization-engine/build.gradle]: For this case I am able to simply add the dependency to the main runtimeClasspath. https://github.com/elastic/elasticsearch/pull/59444/commits/ef3466837535242c2d0c4be28241bde6ff2af951
```
javaRestTestImplementation project.sourceSets.main.runtimeClasspath
```

File: plugins/examples/rest-handler/build.gradle
Comment (jakelandis) [reply plugins/examples/rest-handler/build.gradle]: done https://github.com/elastic/elasticsearch/pull/59444/commits/849d19db73f47c21baaa3cb591d28521bd9e6294

File: plugins/analysis-icu/build.gradle
Comment (jakelandis) [reply plugins/analysis-icu/build.gradle]: done https://github.com/elastic/elasticsearch/pull/59444/commits/849d19db73f47c21baaa3cb591d28521bd9e6294

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Pull request merged**
